### PR TITLE
Adopt esptool & etckeeper

### PR DESCRIPTION
**Pull request by mjeveritt**
Fix Gentoo bug [620042](https://bugs.gentoo.org/620042).

### CODE REVIEW

File: sys-apps/etckeeper/etckeeper-1.18.1.ebuild [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 
 inherit eutils bash-completion-r1 prefix python-r1
 
 DESCRIPTION="A collection of tools to let /etc be stored in a repository"
 HOMEPAGE="https://etckeeper.branchable.com/"
-SRC_URI="https://github.com/joeyh/etckeeper/archive/${PV}.tar.gz -> ${P}.tar.gz"
+SRC_URI="https://git.joeyh.name/index.cgi/etckeeper.git/snapshot/${P}.tar.gz"
 
 LICENSE="GPL-2"
 KEYWORDS="amd64 ~arm x86"
```
Comment (jonasstein) [inline sys-apps/etckeeper/etckeeper-1.18.1.ebuild]: EAPI Bump breaks stabilisation

File: sys-apps/etckeeper/etckeeper-1.18.1.ebuild [hunk 2]
```diff
@@ -1,15 +1,15 @@
-# Copyright 1999-2015 Gentoo Foundation
+# Copyright 1999-2017 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=5
+EAPI=6
 
 PYTHON_COMPAT=( python2_7 )
 
 inherit eutils bash-completion-r1 prefix python-r1
```
Comment (jonasstein) [inline sys-apps/etckeeper/etckeeper-1.18.1.ebuild]: what is eutils needed for?

**Comment by mjeveritt** [comment]
That should be better, thanks.

**Comment by mjeveritt** [comment]
That said, conceivably there should be another commit in there ... >,< :man_facepalming:

File: sys-apps/etckeeper/etckeeper-1.18.1-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+LICENSE="GPL-2"
+KEYWORDS="~amd64 ~arm ~x86"
+SLOT="0"
+IUSE="bazaar cron"
+REQUIRED_USE="bazaar? ( ${PYTHON_REQUIRED_USE} )"
+
+VCS_DEPEND="dev-vcs/git
+	dev-vcs/mercurial
+	dev-vcs/darcs"
+DEPEND="bazaar? ( dev-vcs/bzr )"
```
Comment (mgorny) [inline sys-apps/etckeeper/etckeeper-1.18.1-r1.ebuild]: You sure you don't need PYTHON_USEDEP in here?

File: sys-apps/etckeeper/etckeeper-1.18.1-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+src_compile() {
+	:
+}
+
+src_install(){
+	emake DESTDIR="${ED}" install
+
+	bzr_install() {
+		${PYTHON} ./${PN}-bzr/__init__.py install --root="${ED}" ||
```
Comment (mgorny) [inline sys-apps/etckeeper/etckeeper-1.18.1-r1.ebuild]: PYTHON should be quoted. Also prefer EPYTHON.

File: sys-apps/etckeeper/etckeeper-1.18.1-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+	use bazaar && python_foreach_impl bzr_install
+
+	if use prefix; then
+		doenvd "${FILESDIR}"/99${PN}
+		eprefixify "${ED%/}"/etc/env.d/99${PN}
+	fi
+
+	newbashcomp bash_completion ${PN}
+	dodoc doc/README.mdwn
+	docinto examples
```
Comment (mgorny) [inline sys-apps/etckeeper/etckeeper-1.18.1-r1.ebuild]: Why create a whole directory for one file?

**Comment by flappyports** [comment]
@mjeveritt do you still intend to work on this PR?

**Comment by mjeveritt** [comment]
Fixed.
What would you like done with the existing 1.18.1 and 1.18.6 ebuilds? revbump those too?

**Comment by mjeveritt** [comment]
GLEP66-compatible commit message.

File: sys-apps/etckeeper/etckeeper-1.18.1-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,71 @@
+# Copyright 1999-2018 Gentoo Foundation
```
Comment (mgorny) [inline sys-apps/etckeeper/etckeeper-1.18.1-r1.ebuild]: Looks like you're revbumping an ebuild that is no longer there.

File: sys-apps/etckeeper/Manifest [hunk 7]
```diff
@@ -1 +1,3 @@
+DIST etckeeper-1.18.1.tar.gz 56366 BLAKE2B ac3cff677139b09eb400acc757852b7650c68a1c50bc0fd6a13bc170215d7abf32cbaadda44fa6ce55245d1c225b0ea0890465b353c31fa31724dea92d7f06c9 SHA512 5b5019d3f8c22ee0486c9102d7dbdc7e86921d432df49c15733e5ed23651c2be87b52e295208be691ef1e44a515d960fcf4f812eaca17f2806e2f5055222ca35
```
Comment (mgorny) [inline sys-apps/etckeeper/Manifest]: …and adding a distfile in the wrong commit.

**Comment by mjeveritt** [comment]
Fantastic! :unamused: 

Rinse and repeat ....

**Comment by mjeveritt** [comment]
11th time lucky ...

File: sys-apps/etckeeper/files/etckeeper-1.18.7-gentoo.patch [hunk 8]
```diff
@@ -0,0 +1,78 @@
+From 026fa7c012a8c44cadfa56f605c0671639a6b04c Mon Sep 17 00:00:00 2001
```
Comment (mgorny) [inline sys-apps/etckeeper/files/etckeeper-1.18.7-gentoo.patch]: This file is identical to `etckeeper-1.18.6-gentoo.patch`

Comment (mjeveritt) [reply sys-apps/etckeeper/files/etckeeper-1.18.7-gentoo.patch]: Apparently so, and according to repoman too.

I could refactor the patch naming and associated code if you so desire?

**Comment by mjeveritt** [comment]
Try that .. it's not amazing, but it works...

**Comment by mgorny** [comment]
Yes, do that instead. Also `versionator.eclass` is obsolete.

**Comment by mjeveritt** [comment]
> Yes, do that instead. Also `versionator.eclass` is obsolete.

Ah thanks, I had my suspicions .. EAPI7 or bash functions instead?!

**Comment by mgorny** [comment]
eapi7-ver or bash patterns. But in this context, you don't need anything special.

**Comment by mjeveritt** [comment]
Cheers. Re-pushed.

**Comment by mgorny** [comment]
Looks good now, thanks!

**Comment by mjeveritt** [comment]
Thanks for bearing with :smile:

**Pull request closed without merging**